src/address.c

```diff
@@ -393,3 +394,174 @@ int getDerivedHDAddress(const char* masterkey, uint32_t account, bool ischange,
+ * @param change_level The BIP44 change level flag to generate derived address.
+ * @param mnemonic The mnemonic code words.
+ * @param passphrase The passphrase (optional).
+ * @param p2pkh_pubkey_master The generated master public key.
+ * @param is_testnet The flag denoting which network, 0 for mainnet and 1 for testnet.
+ *
+ * return: 0 (success), -1 (fail)
+ */
+int getDerivedHDAddressFromMnemonic(const uint32_t account, const uint32_t index, const CHANGE_LEVEL change_level, const MNEMONIC mnemonic, const PASSPHRASE pass, char* p2pkh_pubkey, const dogecoin_bool is_testnet) {
+
+    /* Validate input */
+    if (!mnemonic) {
+        fprintf(stderr, "ERROR: Invalid mnemonic\n");
+        return -1;
+    }
+
+    /* Initialize variables */
+    dogecoin_hdnode node;
+    dogecoin_hdnode bip44_key;
+    char keypath[BIP44_KEY_PATH_MAX_LENGTH + 1] = "";
+
+    /* Allocate memory for seed and initialize */
+    uint8_t* seed = malloc(64);
+    if (seed == NULL) {
+        fprintf(stderr, "ERROR: Failed to allocate memory for seed\n");
+        return -1;
+    }
+    memset(seed, 0, 64);
+
+    /* Convert mnemonic to seed */
+    if (dogecoin_seed_from_mnemonic (mnemonic, pass, seed) == -1) {
+        dogecoin_free (seed);
+        return -1;
+    }
+
+    /* Generate the root key from the seed */
+    dogecoin_hdnode_from_seed(seed, 64, &node);
+    dogecoin_free (seed);
+
+    /* Derive the BIP 44 extended key */
+    if (derive_bip44_extended_private_key(&node, account, NULL, change_level, NULL, is_testnet, keypath, &bip44_key) == -1) {
+        return -1;
+    }
+
+    /* Derive the child private key at the index */
+    if (derive_bip44_extended_private_key(&node, account, &index, change_level, NULL, is_testnet, keypath, &bip44_key) == -1) {
+        return -1;
+    }
+
+    /* Generate the address */
+    dogecoin_hdnode_get_p2pkh_address(&bip44_key, is_testnet ? &dogecoin_chainparams_test : &dogecoin_chainparams_main, p2pkh_pubkey, P2PKH_ADDR_STRINGLEN);
+
+   return 0;
+
+}
+
+/**
+ * @brief This function generates a HD master key and p2pkh ready-to-use corresponding dogecoin address from a mnemonic.
+ *
+ * @param wif_privkey_master The generated master private key.
+ * @param p2pkh_pubkey_master The generated master public key.
+ * @param mnemonic The mnemonic code words.
+ * @param passphrase The passphrase (optional).
+ * @param is_testnet The flag denoting which network, 0 for mainnet and 1 for testnet.
+ *
+ * return: 0 (success), -1 (fail)
+ */
+int generateHDMasterPubKeypairFromMnemonic(char* wif_privkey_master, char* p2pkh_pubkey_master, const MNEMONIC mnemonic, const PASSPHRASE pass, const dogecoin_bool is_testnet) {
+
+    /* Validate input */
+    if (!mnemonic) {
+        fprintf(stderr, "ERROR: Invalid mnemonic\n");
+        return -1;
+    }
+
+    /* Initialize variables */
+    dogecoin_hdnode node;
+
+    /* Allocate memory for seed and initialize */
+    uint8_t* seed = malloc(64);
```

Comment:
Same as previous function - Can this just be on the stack as `uint8_t seed[MAX_SEED_SIZE]` or just `SEED seed` to avoid all the worry about allocation/freeing?